Libraries/src/Amazon.Lambda.Annotations/README.md

```diff
@@ -68,37 +187,141 @@ public class Functions
+For each arithmetic operation a separate C# method is added containing the `LambdaFunction` attribute. The `LambdaFunction` attribute
+ensures the dependency injection framework is hooked up to the Lambda function and the Lambda function will be declared in the 
+CloudFormation template. 
 
-## Source Generator
+Since these Lambda functions are responding to API Gateway events the `HttpApi` attribute is added 
+to registered the event source in CloudFormation along with the HTTP verb and resource path. The `HttpApi` attribute also enables 
+mapping of the HTTP request components to method parameters. In this case the operands used for the arithmetic operations are 
+mapped from the resource path. Checkout the list of Lambda attributes in the reference section to see how to map other components
+of the HTTP request to method parameters.
 
-To bridge the gap from Lambda Annotations programming model to the normal programming model a .NET source generator is included in this package.
-After adding the attributes to your .NET code the source generator will generate the translation code between the 2 programming models. It will also
-keep in sync the generated information including a new function handler string into the CloudFormation template. The usage of source
-generator is transparent to the user. The source generator supports syncing of both JSON and YAML CloudFormation templates.
 ```csharp
-[LambdaFunction(Name = "Plus")]
-[RestApi(LambdaHttpMethod.Get, "/plus/{x}/{y}")]
-public int Plus(int x, int y)
+[LambdaFunction()]
+[HttpApi(LambdaHttpMethod.Get, "/add/{x}/{y}")]
+public int Add(int x, int y, ILambdaContext context)
 {
-    return x + y;
+    context.Logger.LogInformation($"{x} plus {y} is {x + y}");
+    return _calculatorService.Add(x, y);
+}
+
+[LambdaFunction()]
+[HttpApi(LambdaHttpMethod.Get, "/substract/{x}/{y}")]
+public int Substract(int x, int y, ILambdaContext context)
+{
+    context.Logger.LogInformation($"{x} substract {y} is {x - y}");
+    return _calculatorService.Substract(x, y);
+}
+
+[LambdaFunction()]
+[HttpApi(LambdaHttpMethod.Get, "/multiply/{x}/{y}")]
+public int Multiply(int x, int y, ILambdaContext context)
+{
+    context.Logger.LogInformation($"{x} multiply {y} is {x * y}");
+    return _calculatorService.Multiply(x, y);
+}
+
+[LambdaFunction()]
+[HttpApi(LambdaHttpMethod.Get, "/divide/{x}/{y}")]
+public int Divide(int x, int y, ILambdaContext context)
+{
+    context.Logger.LogInformation($"{x} divide {y} is {x / y}");
+    return _calculatorService.Divide(x, y);
 }
 ```
-The source generator adds the following entry in the CloudFormation template corresponding to the above Lambda function:
+
+For each `LambdaFunction` declared the source generator will updated the CloudFormation template with the corresponding resource. 
+Here you can see the Lambda CloudFormation resource has the `Handler` property to the generated code by the source generator. 
```

Comment:
Awkward.  "Here you can see that the Handler property has been added to the generated code for the Lambda CloudFormation resource." ?

Reply:
Reworked